scripts/download_artifacts.py

```diff
@@ -0,0 +1,151 @@
+        print(f"testsuite failed for {build_name}")
+        if not build_failed:
+            with open("./logs/failed_testsuite.txt", "a+") as f:
+                f.write(f"{build_name}|Cannot find testsuite artifact\n")
+        return -1
+    return artifact[0].id
+
+def download_artifact(artifact_name: str, artifact_id: str, token: str):
+    """
+    Uses GitHub api endpoint to download and extract the log artifact into
+    directory called ./logs
+    """
+    params = {"Accept": "application/vnd.github+json",
+              "Authorization": f"token {token}",
+              "X-Github-Api-Version": "2022-11-28"}
+    artifact_zip_name = artifact_name.replace(".log", ".zip")
+    print(artifact_zip_name)
+    print(f"downloading: {artifact_id}")
+    r = requests.get(f"https://api.github.com/repos/patrick-rivos/riscv-gnu-toolchain/actions/artifacts/{artifact_id}/zip", headers=params)
+    print(r.status_code)
+    with open(f"./{artifact_zip_name}", "wb") as f:
+        f.write(r.content)
+    with ZipFile(f"./{artifact_zip_name}", "r") as zf:
+        zf.extractall(path=f"./{artifact_name.split('.log')[0]}")
+    os.rename(f"./{artifact_name.split('.log')[0]}/{artifact_name}", f"./logs/{artifact_name}")
+
+def download_all_artifacts(current_hash: str, token: str):
+    """
+    Goes through all possible artifact targets and downloads it
+    if it exists. Downloads previous successful hash's artifact
+    as well. Runs comparison on the downloaded artifacts
+    """
+
+    auth = Auth.Token(token)
+    g = Github(auth=auth)
+
+    repo = g.get_repo("patrick-rivos/riscv-gnu-toolchain")
+
+    prev_commits = gcc_hashes(current_hash, False)
+    artifact_names = get_possible_artifact_names()
+    for artifact_name in artifact_names:
+        artifact = artifact_name.format(current_hash)
+        artifact_id = check_artifact_exists(artifact, repo)
+        if artifact_id == -1:
+            continue
+        # comparison output path
+        compare_path = f"./summaries/{artifact + '-report-summary.md'}"
+        artifact += "-report.log"
+        artifact_name += "-report.log"
+        # download current artifact
+        download_artifact(artifact, str(artifact_id), token)
+
+        # download previous artifact
+        base_hash, base_id = get_valid_artifact_hash(prev_commits, token, artifact_name)
+        if base_hash == "No valid hash":
+            print(f"no baseline for {artifact}")
+            with open("./logs/no_baseline.txt", "a+") as f:
+                f.write(f"{artifact}\n")
+            base_hash = current_hash + "-no-baseline"
+            try:
+                run(base_hash, f"./logs/{artifact}", base_hash, f"./logs/{artifact}", compare_path)
+            except (RuntimeError, ValueError) as err:
+                with open("./logs/failed_testsuite.txt", "a+") as f:
+                    f.write(f"{artifact}|{err}\n")
+            continue
+        download_artifact(artifact_name.format(base_hash), str(base_id), token)
```

Comment:
If we know the base_hash is invalid from the previous if, why do we try downloading it regardless?

Reply:
The previous if checks if `base_hash == "No valid hash"` so if the program enters this if statement, the `continue` would skip the download after comparing the base_hash to itself (line 129). Otherwise, it will download the `base_hash` artifacts

Reply:
Ah that makes more sense. I missed the `continue` and got confused.